src/help/urlUtils.ts

```diff
@@ -0,0 +1,53 @@
+import {PlatformUrlOptions} from '..';
+import {platformUrl} from '../environment';
+
+export interface URLBuilder {
+  baseAPIURLForUpdate: URL;
+  fileContainerUrl: URL;
+}
+
+abstract class BaseUrlBuilder {
+  protected constructor(
+    private organizationId: string,
+    private options: Required<PlatformUrlOptions>
+  ) {}
+  protected get platformURL() {
+    return `${platformUrl(this.options)}/${this.organizationId}`;
+  }
+  public get fileContainerUrl() {
+    return new URL(`${this.platformURL}/files`);
+  }
+  public abstract get baseAPIURLForUpdate(): URL;
+}
+
+export class PushUrlBuilder extends BaseUrlBuilder implements URLBuilder {
```

Comment:
nitpick: this looks a bit over engineered for what could probably have been just simple function(s) ?

Reply:
That is indeed over the top, but that's a consequence for using the Strategy pattern.
Since both `FileContainerStrategy` and `StreamChunkStrategy` require a URLBuilder, I had to define some sort of common interface.

The whole push/stream situation is a bit confusing IMO. Both catalog and push source can use the `FileContainerStrategy` but with different Urls, hence the UrlBuilder.

I don't think I could achieve the same result with functions without duplicating code at several places.